charts/pwa/templates/ingress.yaml

```diff
@@ -1,131 +1,66 @@
 {{- if .Values.ingress.enabled -}}
-{{- $fullName := include "pwa-ingress.service" . -}}
-{{- $svcPort := .Values.cache.service.port -}}
-{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
-  {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
-  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
-  {{- end }}
-{{- end }}
-{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
-apiVersion: networking.k8s.io/v1
-{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
-apiVersion: networking.k8s.io/v1beta1
-{{- else -}}
-apiVersion: extensions/v1beta1
+{{ range $name, $value := .Values.ingress.instances }}
+{{ with $value }}
+{{- $tlsDict := dict -}}
```

Comment:
I got completely lost here 🤷 Can you give me an example of the tls secret handling and how it would look like in the values.yaml  for example if a customer has 5 different hosts, each of them potentially bringing their own secret?

Reply:
I was thinking about adding comments but could only think about YAML `# ...` which would turn up in the resulting charts. Is there a way to comment in this templating language?

The problem is that the way I proposed the tlsSecret on top and the tlsSecret override per host is an inverse mapping than what the kubernetes chart wants. So in this part we create a map with key tlsSecret and a list of assigned hosts as value. It iterates through all hosts and assigns them to the correct secret. Later on, if there are any keys in the map, the tls section of the Ingress is rendered simply by iterating over the map.

```yaml
    ingress:
      tlsSecretName: pwa-secret
      hosts:
        - host: pwa.example.local
          tlsSecretName: pwa-secret2
        - host: pwa2.example.local
          tlsSecretName: pwa-secret2
        - host: pwa3.example.local
        - host: pwa4.example.local
        - host: pwa5.example.local
        - host: pwa6.example.local
          tlsSecretName: pwa-secret3
```

gives you

```yaml
  tls:
    - secretName: pwa-secret
      hosts:
        - "pwa3.example.local"
        - "pwa4.example.local"
        - "pwa5.example.local"
    - secretName: pwa-secret2
      hosts:
        - "pwa.example.local"
        - "pwa2.example.local"
    - secretName: pwa-secret3
      hosts:
        - "pwa6.example.local"
```

Reply:
I also added some comments now 👍

Reply:
Errrm it is 2 different things :D For the Helm templating you can add comments like you did. YAML on itself does not support comments per its definition, so after the deployment, any comments will be stripped and removed.

But thanks! Now it makes more sense!